sdk/python/feast/infra/offline_stores/remote.py

```diff
@@ -79,34 +68,51 @@ def _to_df_internal(self, timeout: Optional[int] = None) -> pd.DataFrame:
     # Invoked to synchronously execute the underlying query and return the result as an arrow table
     # This is where do_get service is invoked
     def _to_arrow_internal(self, timeout: Optional[int] = None) -> pa.Table:
-        # Generate unique command identifier
-        command_id = str(uuid.uuid4())
-        command = {
-            "command_id": command_id,
-            "api": "get_historical_features",
-            "feature_view_names": self.feature_view_names,
-            "name_aliases": self.name_aliases,
-            "feature_refs": self.feature_refs,
-            "project": self.project,
-            "full_feature_names": self._full_feature_names,
-        }
-        command_descriptor = fl.FlightDescriptor.for_command(
-            json.dumps(
-                command,
-            )
+        return _send_retrieve_remote(
+            self.api, self.api_parameters, self.entity_df, self.table, self.client
         )
 
-        self._put_parameters(command_descriptor)
-        flight = self.client.get_flight_info(command_descriptor)
-        ticket = flight.endpoints[0].ticket
-
-        reader = self.client.do_get(ticket)
-        return reader.read_all()
-
     @property
     def on_demand_feature_views(self) -> List[OnDemandFeatureView]:
         return []
 
+    @property
+    def metadata(self) -> Optional[RetrievalMetadata]:
+        return self._metadata
+
+    @property
+    def full_feature_names(self) -> bool:
+        logger.info(f"{self.api_parameters}")
+        return self.api_parameters["full_feature_names"]
+
+    def persist(
```

Comment:
please add a comment to simplify the community reviewers, to clarify that we're using an Arrow action to execute the persist remotely